### PR TITLE
Add tile_images

### PR DESCRIPTION
**Pull request by yuyu2172**
Merge after #421

```python
from chainercv.visualizations import vis_image
from chainercv.datasets import VOCSemanticSegmentationDataset
from chainercv.utils import make_grid
import matplotlib.pyplot as plt
from chainercv.transforms import center_crop
import numpy as np

dataset = VOCSemanticSegmentationDataset()
imgs = np.array([center_crop(data[0], (224, 224)) for data in dataset[:10]])
grid = make_grid(imgs, n_col=3, pad=3)
vis_image(grid)
plt.show()
```

Output (with pad=3)
figure_1-3

Output (with pad=50)
figure_1-4

### CODE REVIEW

**Comment by Hakuyume** [comment]
How about `tile_images`? `make_grid` sounds ambiguous.

**Comment by yuyu2172** [comment]
OK

File: chainercv/utils/__init__.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 from chainercv.utils.download import extractall  # NOQA
 from chainercv.utils.image import read_image  # NOQA
 from chainercv.utils.image import write_image  # NOQA
+from chainercv.utils.image.tile_images import tile_images  # NOQA
```
Comment (Hakuyume) [inline chainercv/utils/__init__.py]: `from chainercv.utils.image  import tile_images`

Comment (Hakuyume) [reply chainercv/utils/__init__.py]: `tile_images` comes before `write_image` in alphabetical order.

File: chainercv/utils/image/tile_images.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+             (W + pad) * n_col)
+    tile = np.empty(shape, dtype=imgs.dtype)
+    tile[:] = np.array(fill).reshape((-1, 1, 1))
+
+    k = 0
+    for y in range(n_row):
+        for x in range(n_col):
+            if k >= B:
+                break
+            start_y = y * (H + pad) + pad // 2 + 1
```
Comment (Hakuyume) [inline chainercv/utils/image/tile_images.py]: Why do you add `1`? In the case that `pad == 0`, the top-left corner of the first image will be `(1, 1)` although it should be `(0, 0)`.

File: chainercv/utils/image/tile_images.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        n_col (int): Number of columns in a tile.
+        pad (int): Amount of pad. Default is 2.
+        fill (float, tuple or ~numpy.ndarray): The value of padded pixels.
+            If it is :class:`numpy.ndarray`,
+            its shape should be :math:`(C, 1, 1)`,
+            where :math:`C` is the number of channels of :obj:`img`.
+
+    Returns:
+        ~numpy.ndarray:
+        An image array in CHW format.
```
Comment (Hakuyume) [inline chainercv/utils/image/tile_images.py]: How about adding a note about the output size?
`The size of this image is :math:((H + pad) * \lceil B / n_col \rceil, (W + pad) * n_col)`

File: chainercv/utils/image/tile_images.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import math
+import numpy as np
+
+
+def tile_images(imgs, n_col, pad=2, fill=0):
+    """Make a tile of images
+
+    Args:
+        imgs (numpy.ndarray): A batch of images whose shape is BCHW.
+        n_col (int): Number of columns in a tile.
```
Comment (Hakuyume) [inline chainercv/utils/image/tile_images.py]: `The number`? (I found both `Number of` and `The number of` in our code)

File: tests/utils_tests/image_tests/test_tile_images.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import unittest
+
+from chainer import testing
+
+from chainercv.utils import tile_images
+
+
+@testing.parameterize(*testing.product({
+    'fill': [128, (104, 117, 123), np.random.uniform(255, size=(3, 1, 1))],
+    'pad': [1, 2, 3]
```
Comment (Hakuyume) [inline tests/utils_tests/image_tests/test_tile_images.py]: How about testing the case `pad = 0`? With this test, [this bug](https://github.com/chainer/chainercv/pull/422/files/111bdd0fe06e3233a1edf59d7d0ac849b5407d75#r139073765) can be found.

**Pull request merged**